app/grandchallenge/algorithms/models.py

```diff
@@ -419,6 +436,23 @@ def create_result(self, *, result: dict):
             )
             self.outputs.add(output_civ)
 
+    @property
+    def rendered_result_text(self):
+        interface = ComponentInterface.objects.get(slug="results-json-file")
+        try:
+            output = self.outputs.get(interface=interface)
+            template_output = JINJA_ENGINE.from_string(
+                self.algorithm_image.algorithm.result_template
+            ).render(result_dict=output.value)
+
+            return md2html(template_output)
+        except UndefinedError as e:
+            return f"Jinja template is incorrect: {e}"
+        except SecurityError:
+            return "Jinja template is not allowed"
+        except ObjectDoesNotExist:
+            return ""
```

Comment:
```suggestion
        try:
            output = self.outputs.get(interface=interface)
        except ObjectDoesNotExist:
            return ""

        try:
            template_output = JINJA_ENGINE.from_string(
                self.algorithm_image.algorithm.result_template
            ).render(result_dict=output.value)
        except (UndefinedError, SecurityError):
            return "Jinja template is invalid"

        return md2html(template_output)
```

Sorry - this is what I meant, this way we can see where we expect the exceptions to be raised.

Reply:
Also, we should avoid giving raw exceptions back to users, this has caused issues in the past, so, I've changed both messages to `"Jinja template is invalid"`